apps/trading/client-pages/portfolio/account-history-container.tsx

```diff
@@ -0,0 +1,245 @@
+import {
+  addDecimal,
+  fromNanoSeconds,
+  t,
+  ThemeContext,
+} from '@vegaprotocol/react-helpers';
+import { useVegaWallet } from '@vegaprotocol/wallet';
+import compact from 'lodash/compact';
+import type { ChangeEvent } from 'react';
+import { useContext } from 'react';
+import { useMemo, useState } from 'react';
+import type { AccountHistoryQuery } from './__generated__/AccountHistory';
+import { useAccountsWithBalanceQuery } from './__generated__/AccountHistory';
+import { useAccountHistoryQuery } from './__generated__/AccountHistory';
+import * as Schema from '@vegaprotocol/types';
+import type { AssetFieldsFragment } from '@vegaprotocol/assets';
+import { useAssetsDataProvider } from '@vegaprotocol/assets';
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+  Splash,
+  Toggle,
+} from '@vegaprotocol/ui-toolkit';
+import { AccountTypeMapping } from '@vegaprotocol/types';
+import { PriceChart } from 'pennant';
+import 'pennant/dist/style.css';
+
+enum DateRange {
+  RANGE_1D = '1D',
+  RANGE_7D = '7D',
+  RANGE_1M = '1M',
+  RANGE_3M = '3M',
+  RANGE_1Y = '1Y',
+  RANGE_YTD = 'YTD',
+  RANGE_ALL = 'All',
+}
+
+const calculateStartDate = (range: string): string | undefined => {
+  const now = new Date();
+  switch (range) {
+    case DateRange.RANGE_1D:
+      return new Date(now.setDate(now.getDate() - 1)).toISOString();
+    case DateRange.RANGE_7D:
+      return new Date(now.setDate(now.getDate() - 7)).toISOString();
+    case DateRange.RANGE_1M:
+      return new Date(now.setMonth(now.getMonth() - 1)).toISOString();
+    case DateRange.RANGE_3M:
+      return new Date(now.setMonth(now.getMonth() - 3)).toISOString();
+    case DateRange.RANGE_1Y:
+      return new Date(now.setFullYear(now.getFullYear() - 1)).toISOString();
+    case DateRange.RANGE_YTD:
+      return new Date(now.setMonth(0)).toISOString();
+    default:
+      return undefined;
+  }
+};
+
+const dateRangeToggleItems = [
```

Comment:
suggestion (non blocking): You might want to change the date range enum to a const object, then we can get the actual values _and_ types from it and have all date ranges defined in a single place (without runtime enum oddities)

```
const DateRange = {
  RANGE_1D: '1D'
  ...
} as const

const dateRangeToggleItmes = Object.entries(([key, value]) => ({
  label: t(key),
  value
})

// for types you can then create them off the object
type Values = typeof DateRange[keyof typeof DateRange];
type Keys = keyof typeof DateRange
```

Reply:
nice one, updated in last commit.